### PR TITLE
Fixing embedding selection issue in experimental glove

### PR DESCRIPTION
**Pull request by Nayef211**
## Description
- Fixing issue where the first file path being extracted from downloaded GloVe `zip` is loaded by default
- Now the correct file will be loaded based on the specified `dim` parameter 
- Refer to #843

### CODE REVIEW

File: torchtext/experimental/vectors.py [hunk 1]
```diff
@@ -103,6 +105,7 @@ def GloVe(name="840B", unk_tensor=None, root='.data', validate_file=True):
             - 840B
             - twitter.27B
             - 6B
+        dim (int): the dimension for the GloVe dataset to load.
```
Comment (zhangguanheng66) [inline torchtext/experimental/vectors.py:108]: Let's be more explicit about the dim options for each category.

File: torchtext/experimental/vectors.py [hunk 2]
```diff
@@ -120,14 +123,16 @@ def GloVe(name="840B", unk_tensor=None, root='.data', validate_file=True):
...
-        '6B': 'https://nlp.stanford.edu/data/glove.6B.zip',
+        "42B": "https://nlp.stanford.edu/data/glove.42B.300d.zip",
+        "840B": "https://nlp.stanford.edu/data/glove.840B.300d.zip",
+        "twitter.27B": "https://nlp.stanford.edu/data/glove.twitter.27B.zip",
+        "6B": "https://nlp.stanford.edu/data/glove.6B.zip",
     }
 
     url = urls[name]
-    cached_vectors_file_path = os.path.join(root, os.path.splitext(os.path.basename(url))[0] + '.pt')
+    file_name = "glove.{}.{}d.txt".format(name, str(dim))
```
Comment (zhangguanheng66) [inline torchtext/experimental/vectors.py:133]: Should we check if dim is support for the specific name?

**Comment by zhangguanheng66** [comment]
Fix #843

File: torchtext/experimental/vectors.py (hunk 1 above)
Comment (Nayef211) [reply torchtext/experimental/vectors.py:108]: Sure. I'll list out what the possible options are.

File: torchtext/experimental/vectors.py (hunk 2 above)
Comment (Nayef211) [reply torchtext/experimental/vectors.py:133]: Do you think I should have a dict like this? And then check that the file name exists within the dict? https://github.com/pytorch/text/blob/fd764ad02968e6310d28ee3c496e1d1d1157aa3e/torchtext/vocab.py#L523

**Comment by zhangguanheng66** [comment]
The `binary_macos_conda_py3.8` test fails again?

**Comment by Nayef211** [comment]
> The `binary_macos_conda_py3.8` test fails again?

Yup this seems to be happening for all my PRs. Could it be another issue on Pytorch's side?

**Comment by zhangguanheng66** [comment]
> > The `binary_macos_conda_py3.8` test fails again?
> 
> Yup this seems to be happening for all my PRs. Could it be another issue on Pytorch's side?
Very likely.

**Pull request merged**